src/ServiceControl.Transports.ASBS/AzureQuery.cs

```diff
@@ -244,23 +234,67 @@ public override async IAsyncEnumerable<QueueThroughput> GetThroughputPerDay(IBro
         }
     }
 
+    async Task<MetricsClient> InitializeMetricsClient(CancellationToken cancellationToken = default)
+    {
+        if (resourceId is null || armClient is null || credential is null)
+        {
+            throw new InvalidOperationException("AzureQuery has not been initialized correctly.");
+        }
+
+        var serviceBusNamespaceResource = await armClient
+            .GetServiceBusNamespaceResource(resourceId).GetAsync(cancellationToken)
+                ?? throw new Exception($"Could not find ServiceBus with resource Id: \"{resourceId}\"");
+
+        // Determine the region of the namespace
+        var regionName = serviceBusNamespaceResource.Value.Data.Location.Name;
+
+        // Build the regional Azure Monitor Metrics endpoint from the audience
+        var metricsEndpoint = BuildMetricsEndpoint(metricsClientAudience, regionName);
+
+        // CreateNewOnMetadataUpdateAttribute the MetricsClient for this namespace
+        return new MetricsClient(
+            metricsEndpoint,
+            credential!,
+            new MetricsClientOptions
+            {
+                Audience = metricsClientAudience,
+                Transport = new HttpClientTransport(
+                    new HttpClient(new SocketsHttpHandler
+                    {
+                        PooledConnectionIdleTimeout = TimeSpan.FromMinutes(2)
+                    }))
+            });
+    }
+
+    static Uri BuildMetricsEndpoint(MetricsClientAudience audience, string regionName)
+    {
+        var region = regionName.ToLowerInvariant();
+        var builder = new UriBuilder(audience.ToString());
+        builder.Host = $"{region}.{builder.Host}";
+        return builder.Uri;
+    }
+
     async Task<IReadOnlyList<MetricValue>> GetMetrics(string queueName, DateOnly startTime, DateOnly endTime,
         CancellationToken cancellationToken = default)
     {
-        var response = await client!.QueryResourceAsync(resourceId,
-            new[] { "CompleteMessage" },
-            new MetricsQueryOptions
+        metricsClient ??= await InitializeMetricsClient(cancellationToken);
+
+        var response = await metricsClient.QueryResourcesAsync(
+            [new ResourceIdentifier(resourceId!)],
```

Comment:
`resourceId` is already a `ResourceIdentifier`. Was there a reason to create a new one?